WaterLog_Backend/Models/DatabaseContext.cs

```diff
@@ -21,6 +21,6 @@ public DatabaseContext(DbContextOptions<DatabaseContext> options) : base(options
         public DbSet<ReadingsEntry> Readings { get; set; }
         public DbSet<SegmentsEntry> Segments { get; set; }
         public DbSet<SegmentEventsEntry> SegmentEvents { get; set; }
-        public DbSet<SegmentLeaksEntry> SegmentLeaks { get; set; }       
+        public virtual DbSet<SegmentLeaksEntry> SegmentLeaks { get; set; }       
```

Comment:
Why is this virtual? Please explain

Reply:
To allow for mocking the database context .

Reply:
I am not 100% sure I 📦 understand. Could you give me an example of where the database must be mocked and how it gets mocked? :)

Reply:
I mocked the database to test for a get API. Maybe I got the concept wrong, but the idea was to tests for the mocked data instead the actual data.